### PR TITLE
README: describe version matrix and branching strategy

### PR DESCRIPTION
**Pull request by guggero**
We add a README that explains how the version compatibility and
branching strategy work.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Golang client library for lnd
+
+`lndclient` is a golang native wrapper for `lnd`'s gRPC interface.
+
+## Compatibility matrix
+
+This library depends heavily on `lnd` for obvious reasons. To support backward
+compatibility with older versions of `lnd`, we use different branches for
+different versions. But there are two "levels" of depending on a version of
```
Comment (bhandras) [inline README.md:9]: nit: I think the "But" is not needed.

**Review by bhandras — APPROVED** [review]
LGTM, nice to have these rules documented!

**Review by carlaKC — APPROVED** [review]
Thanks for writing this up @guggero!

Not sure how useful PR templates are to other folks, but it could be good to have a checklist item that's just a reminder about checking versions?

**Comment by guggero** [comment]
Good idea, added a PR template.

**Pull request merged**